src/collector/facet_collector.rs

```diff
@@ -789,6 +807,15 @@ mod tests {
         );
         Ok(())
     }
+
+    #[test]
+    fn is_child_facet() {
+        assert!(super::is_child_facet(&b"foo"[..], &b"foo\0bar"[..]));
```

Comment:
Can you rename it to `is_child_or_root_facet`

Reply:
You mean the test or the original function `is_child_facet`?

In case of the original function, I think the suggest name is a bit confusing because it supposedly returns true if the given facet is a child facet, i.e. information on the possible child facet, not the parent facet. `is_parent_or_root_facet` would be better if the case should be noted explicitly in the name but then again every facet is a child facet of the root facet, so the existing name fits IMHO, just the implementation lacks the one crucial check.

Reply:
ah yes, you are right, I misunderstood how the function is used